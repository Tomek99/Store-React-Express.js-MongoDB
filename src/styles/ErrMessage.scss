.errorComponent {
    height: 1rem;
    font-size: 1.2rem;
    color: red;
    font-weight: 500;
}
