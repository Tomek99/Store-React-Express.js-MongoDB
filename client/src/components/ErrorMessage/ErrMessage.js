import React from "react";
import { ErrorMessage } from "formik";
import styles from "./ErrMessage.module.scss";
import PropTypes from "prop-types";

function ErrMessage({ name }) {
  return (
    <ErrorMessage name={name}>
      {(msg) => <div className={styles.errorText}>{msg}</div>}
    </ErrorMessage>
  );
}

ErrMessage.propTypes = {
  name: PropTypes.string,
};
export default ErrMessage;
