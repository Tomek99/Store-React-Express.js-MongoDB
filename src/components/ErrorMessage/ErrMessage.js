import React from "react";
import { ErrorMessage } from "formik";
import styles from "./ErrMessage.module.scss";

function ErrMessage(pros) {
  return (
    <div className={styles.errorComponent}>
      <ErrorMessage name={pros.name} />
    </div>
  );
}

export default ErrMessage;
