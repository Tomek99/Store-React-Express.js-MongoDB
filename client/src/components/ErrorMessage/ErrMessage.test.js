import { render } from "@testing-library/react";
import ErrMessage from "./ErrMessage";
import { Formik } from "formik";

test("should render errMessage component", () => {
  const data = "number";
  const { container } = render(
    <Formik initialValues={{}} onSubmit={() => {}}>
      <ErrMessage name={data} />
    </Formik>
  );

  expect(container).toMatchSnapshot();
});
