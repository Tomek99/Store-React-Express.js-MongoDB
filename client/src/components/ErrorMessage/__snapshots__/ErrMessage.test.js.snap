// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render errMessage component 1`] = `<div />`;
